modules/widgets/migrate-to-core/gallery-widget.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * Migration from Jetpack's Gallery Widget to WordPress' Core Gallery Widget.
+ *
+ * @since 5.5
+ *
+ * @package Jetpack
+ */
+/**
+ * Migrates all active instances of Jetpack's Gallery widget to Core's Media Gallery widget.
+ */
+function jetpack_migrate_gallery_widget() {
+	// Only trigger the migration from wp-admin and outside unit tests
+	if ( ! is_admin() || defined( 'PHPUNIT_JETPACK_TESTSUITE' ) ) {
+		return;
+	}
+
+	// Only migrate if the new widget is available and we haven't yet migrated
+	if ( ! class_exists( 'WP_Widget_Media_Gallery' ) || Jetpack_Options::get_option( 'gallery_widget_migration' ) ) {
+		return;
+	}
+
+	$old_widgets = get_option( 'widget_gallery', array() );
+	$media_gallery = get_option( 'widget_media_gallery', array() );
+	$sidebars_widgets = wp_get_sidebars_widgets();
+
+	// Array to store legacy widget ids in to unregister on success.
+	$widgets_to_unregister = array();
+
+	$old_widgets = array_filter( $old_widgets, 'jetpack_migrate_gallery_widget_is_importable' );
```

Comment:
You should store this value somewhere in case we find an error in this script down the line.  That way we have a record of the original value to reference.  This will also allow us to periodically check the results vs. og value to ensure we're seeing the expected migration behavior.

Probably also the og sidebar is good to store as well.

Reply:
With 02bc7df, the original `widget_gallery`option is no longer deleted.